README.md

```diff
@@ -237,6 +268,210 @@ You can do the same for Kafka Streams and Kafka, by using the `kstreams` and `ka
 
 Lastly, create a Makefile in the `code` directory to invoke the harness runner and check any outputs that it produces. Then modify the `.semaphore/semaphore.yml` file to invoke that Makefile. This will make sure your tutorial gets checked by the CI system.
 
+## Testing Locally
+
+Given the test harness is the `heart` of a tutorial, it will be helpful to describe in detail how to work with a `kafka|ksql|kstreams.yml` file.  It should be noted the harness file is in the [YAML file formt](https://en.wikipedia.org/wiki/YAML), so formatting properly is essential.  The harness files generates the structure of the rendered tutorial and also validates any output of tutorial steps against expected values.
+
+This section is not meant to be an exhaustive decription of the harness file.  New tutorial authors should not need to create a harness file from scratch, using either the `tools/gen_project.sh` or  `tools/clone.sh` script will provide a usable harness file.  This section should provide enough guidance to add, update, or remove sections as needed.
+
+#### TL;DR
+
+To run a tutorial programatically do the following stepes. Note to follow these instructions you'll need to checkout the kafka-tutorials repo:
+
+1. git clone https://github.com/confluentinc/kafka-tutorials.git
+2. cd kafka-tutorials
```

Comment:
Should we add a 3rd step to setup environment per the instructions at https://github.com/confluentinc/kafka-tutorials#setup ?  If so, that might be where we provide the Dockerfile solution?

Reply:
Or is it the case that running the harness_runner does NOT require all those other dependencies?

Reply:
Just wanted to document a suggestion from @vdesabou here -- this differentiation would address the confusion I had and help other users as well.

> It would be great to have a clarification in prerequisite section to specify that dependencies are only to build the site and not required to test locally.